src/util/Platform.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Utils for Platform.ts.
+ *
+ * @license
+ * Copyright 2023 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { networkInterfaces } from 'os';
+
+export class Platform {
+  readonly interfaces = networkInterfaces();
+
+  constructor() {
+    this.listInterfaces()
+  }
+
+  private listInterfaces() {
+    const { interfaces } = this
+
+    console.log('Select a network interface:');
+
+    Object.keys(interfaces).forEach((interfaceName, index) => {
+      const addresses = interfaces[interfaceName]!.map((addressInfo) => addressInfo.address);
+      console.log(`${index + 1}: ${interfaceName} (${addresses.join(', ')})`);
+    });
+  }
+
+  selectInterface(): Promise<string> {
```

Comment:
async ?

You can use getPromiseResolver to easily convert stdin.on into an aync method.

const { promise, resolver } = await getPromiseResolver<[data type]>();
stdin.on('data', data => resolver(data));
const data = await promise;
stdin.removeAllListeners('data');